fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -817,12 +817,12 @@ private CompletableFuture<Void> runSyntheticMaintainers(@Nonnull Map<RecordType,
     @API(API.Status.EXPERIMENTAL)
     @Nonnull
     @Override
-    public CompletableFuture<FDBSyntheticRecord> loadSyntheticRecord(@Nonnull Tuple primaryKey) {
+    public CompletableFuture<FDBSyntheticRecord> loadSyntheticRecord(@Nonnull Tuple primaryKey, final IndexOrphanBehavior orphanBehavior) {
```

Comment:
We should add some tests of this method with the two types of synthetic record types we provide.
I see there are calls in `UnnestedRecordTypeTest`. Unfortunately, I don't see any similar tests for joined record types. It might also make sense to do tests of `loadByPrimarkyKeyAsync` for the two types directly, rather than via this method.

Reply:
Done: Added tests for both `UnnestedRecordType` and `JoinedRecordType` for loading records when constituents are missing.